test/words/test/TestINodeAssign.java

```diff
@@ -0,0 +1,36 @@
+package words.test;
+import static org.junit.Assert.*;
+import words.ast.*;
+import words.exceptions.*;
+import words.environment.*;
+
+import org.junit.Test;
+
+public class TestINodeAssign extends TestINode {
+	
+	@Test
+	public void assignProperty() throws WordsRuntimeException {
+		WordsObject alexObject = environment.createObject("Alex", "thing", new WordsPosition(0,0));
+		LNodeReference alexLNodeRef = new LNodeReference("Alex's");
+		INodeReferenceList alexRefList = new INodeReferenceList(alexLNodeRef);
+		
+		WordsObject bobObject = environment.createObject("Bob", "thing", new WordsPosition(0,0));
+		LNodeIdentifier bobIdentifier = new LNodeIdentifier("Bob");
+		
+		INodeAssign numAssign = new INodeAssign(alexRefList, new LNodeIdentifier("height"), fiveLeaf);
+		INodeAssign stringAssign = new INodeAssign(alexRefList, new LNodeIdentifier("greeting"), stringLeaf);
+		INodeAssign objAssign = new INodeAssign(alexRefList, new LNodeIdentifier("friend"), new INodeReferenceList(), bobIdentifier);
+		INodeAssign nothingAssign = new INodeAssign(alexRefList, new LNodeIdentifier("myNothing"), nothingLeaf);
+		
+		
+		numAssign.eval(environment);
+		stringAssign.eval(environment);
+		objAssign.eval(environment);
+		nothingAssign.eval(environment);
+		
+		assertEquals("Number assignment successful", alexObject.getProperty("height").numProperty, 5.0, .0001);
```

Comment:
It's often best practice to separate out these tests -- create four different tests, one for String, one for number, one for object, and one for nothing. That way, if one of these tests fail, we only see that specific functionality has failed. 

Line numbers will sort of do this, but will depend on what fails first. I.e. if number assignment failed, then we have no idea if string assignment works, because the program failed before it reached there.